File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> count;

        //counting
        for(int num: nums) {
            count[num]++;
        }

        //storing in pair
        vector<pair<int, int>> ans;
        for(auto& p: count) {
            ans.push_back({p.second, p.first});
        }

        //sorting in descending
        sort(ans.rbegin(), ans.rend());

        //top k
        vector<int> result;
        for(int i = 0; i < k; i++) {
            result.push_back(ans[i].second);
        }
        return result;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """

        #counting
        count = {}
        for num in nums:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        
        #paring
        freq_pair = []
        for num in count:
            freq_pair.append((count[num], num))

        #sort
        freq_pair.sort(reverse = True)

        #top k
        result = []
        for i in range(k):
            result.append(freq_pair[i][1])
            
        return result